diff -ru /usr/share/lipstick-jolla-home-qt5/compositor.qml /usr/share/lipstick-jolla-home-qt5/compositor.qml
--- /usr/share/lipstick-jolla-home-qt5/compositor.qml	2022-06-05 17:18:18.153375000 +0200
+++ /usr/share/lipstick-jolla-home-qt5/compositor.qml	2022-08-15 20:21:15.026479097 +0200
@@ -21,6 +21,7 @@
 import Sailfish.Silica.private 1.0
 import Sailfish.Lipstick 1.0
 import com.jolla.lipstick 0.1
+import com.mousetrackeritem 1.0
 
 import "layers"
 import "launcher"
@@ -1328,41 +1329,38 @@
         }
     }
 
-    MouseTracker {
-        id: mouseTracker
-
-        anchors.fill: parent
-        enabled: displayCursor.value && available && !lipstickSettings.lowPowerMode
-        rotation: QtQuick.Screen.angleBetween(Lipstick.compositor.topmostWindowOrientation, QtQuick.Screen.primaryOrientation)
-
-        onAvailableChanged: if (available) mouseVisibilityTimer.restart()
-        onMouseXChanged: if (available) mouseVisibilityTimer.restart()
-        onMouseYChanged: if (available) mouseVisibilityTimer.restart()
-
-        Timer {
-            id: mouseVisibilityTimer
-
-            // Hide after 10 minutes of idle
-            interval: 10 * 60 * 1000
-        }
-
-        ConfigurationValue {
-            id: displayCursor
-            key: "/desktop/sailfish/compositor/display_cursor"
-            defaultValue: false
-        }
-
-        Image {
-            // JB#56057: Support custom pointer graphics with different hotspot co-ordinates
-            // Now the hotspot co-ordinates below need to be updated if graphic-pointer-default icon is changed
-            property real hotspotX: 13/48 * width
-            property real hotspotY: 4/48 * height
-            x: mouseTracker.mouseX - hotspotX
-            y: mouseTracker.mouseY - hotspotY
-            opacity: mouseTracker.enabled && mouseVisibilityTimer.running ? 1.0 : 0.0
-            Behavior on opacity { FadeAnimator {}}
-            source: "image://theme/graphic-pointer-default"
-        }
+    MouseTrackerItem {
+        id: mouseTrackerItem
+        target: root
+        onMiddleButtonPressed: {
+            if (root.appLayer.active || root.homeLayer.events.active)
+                root.goToSwitcher(true)
+            else
+                root.goToEvents()
+        }
+        // only wheel events are delivered when screen is locked
+//        onWheelRolled: {
+//            if (lipstickSettings.lockscreenVisible) {
+//                console.log("unlock from wheel event")
+//                root.unlock()
+//                //FIXME: it's not really unlocked until touch event is generated
+//            }
+//        }
+    }
+
+    // cursor
+    // there is graphic-pointer-default now
+    Rectangle {
+        width: 20 * Theme.pixelRatio
+        height: width
+        radius: width/2
+        visible: mouseTrackerItem.mouseActive && !lipstickSettings.lockscreenVisible
+        x: mouseTrackerItem.mouseX - width/2
+        y: mouseTrackerItem.mouseY - height/2
+        z: 20
+        opacity: 0.7
+        color: mouseTrackerItem.pressed ? "#828282" : "white"
+        border.color: "black"
     }
 
     Component {
